### DISCUSSION TITLE
Compilation failed. Receiving ParserError

### DISCUSSION THREAD

**Discussion by daryle-tan** [ConsenSys-archive/truffle#5278] [General Q&A]
I am currently doing the truffle-pet-shop-tutorial and after running $ truffle compile, I receive this error message: ParsedContract.sol:7:1: ParserError: Expected pragma, import directive or contract/interface/library definition.
function adopt(uint petId) public returns (uint) {
^------^
Compilation failed. See above.

**Comment by kevinbluer** [comment]
Hey @daryle-tan! Any chance you might be able to share the full code (perhaps via a [Github gist](https://gist.github.com/)) so we can better help troubleshoot?

**Comment by daryle-tan** [comment]
Hi Kevin,

Thank you for the response. Here is the Github gist <script src="
https://gist.github.com/daryle-tan/2f456998fac42cd2ecf96a64aeb7dde3.js"></script>
or https://gist.github.com/2f456998fac42cd2ecf96a64aeb7dde3.git

On Mon, Jul 11, 2022 at 1:21 PM Kevin Bluer ***@***.***>
wrote:

> Hey @daryle-tan <https://github.com/daryle-tan>! Any chance you might be
> able to share the full code (perhaps via a Github gist
> <https://gist.github.com/>) so we can better help troubleshoot?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/trufflesuite/discussions/5278#discussioncomment-3123561>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AWX2QVA7632MMWVTNITN2ULVTRQ3TANCNFSM53D7RRZA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Daryle Tan
